cpp/dolfinx/fem/Function.h

```diff
@@ -397,29 +392,22 @@ class Function
 
     // Interpolate point values on each cell (using last computed value if
     // not continuous, e.g. discontinuous Galerkin methods)
-    Eigen::Array<double, Eigen::Dynamic, 3, Eigen::RowMajor> x(num_dofs_g, 3);
-    Eigen::Array<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> values(
-        num_dofs_g, value_size_loc);
     auto map = mesh->topology().index_map(tdim);
     assert(map);
     const std::int32_t num_cells = map->size_local() + map->num_ghosts();
+
+    Eigen::Array<int, Eigen::Dynamic, 1> cells(x_g.rows());
```

Comment:
Use `std::vector` here. Especially for 1D arrays used mainly as just data structures we want to move away from Eigen.